pennylane/circuit_graph.py

```diff
@@ -116,6 +117,58 @@ def __init__(self, ops, variable_deps):
                 # Create an edge between this and the previous operator
                 self._graph.add_edge(wire[i - 1], wire[i])
 
+    def create_hash_argument(self):
```

Comment:
Maybe there is a better name for this method that is more descriptive... maybe `serialize`? Since that is what this method is doing --- we are essentially just serializing all the information about the circuit graph

Reply:
I think you are right. One thing about calling it as ``serialization``, is that it might imply the process of sending the serialized data and later coming to a ``deserialization`` process. Although this might not be happening here, as the ``hashed`` value of the serialized data is used later, there is no reason for not calling it so. As a matter of fact, if such a feature was to be added to ``CircuitGraph``, then this ``serialize`` method could be used as the first step. Currently, however, trying to recreate the circuit as in trying to deserialize the data merely [from the ``hashed`` value would be really difficult](https://stackoverflow.com/questions/3006710/python-unhash-value)). Good thing this is not the purpose for which we would like to use it now. :)

Reply:
This was kind of my motivation for calling it serialize 😆 We might have to start thinking about how we serialize/deserialize a QNode in the near future, which will involve serializing both the circuit graph and the device.